include/rmm/mr/device/system_memory_resource.hpp

```diff
@@ -94,8 +94,12 @@ class system_memory_resource final : public device_memory_resource {
    */
   void* do_allocate(std::size_t bytes, [[maybe_unused]] cuda_stream_view stream) override
   {
-    return rmm::detail::aligned_host_allocate(
-      bytes, CUDA_ALLOCATION_ALIGNMENT, [](std::size_t size) { return ::operator new(size); });
+    try {
+      return rmm::detail::aligned_host_allocate(
+        bytes, CUDA_ALLOCATION_ALIGNMENT, [](std::size_t size) { return ::operator new(size); });
+    } catch (std::bad_alloc const&) {
```

Comment:
What is the purpose of this, is it just changing the exception type?  I think you should try to incorporate the `what()` from the `bad_alloc` into the new exception.

Reply:
Done.